File: labs/google_maps_files/make_google_map.m
headerFile = 'header.txt';
footerFile = 'footer.txt';

outputFile = 'mymap.html';

mymap = fopen(outputFile,'w');

header = fopen(headerFile);
while ~feof(header)
  headerLine = fgetl(header);
  fprintf(mymap,'%s\n',headerLine);
end
fclose(header);

fprintf(mymap,'var marker = new google.maps.Marker({\n');
fprintf(mymap,'   position: new google.maps.LatLng(42.3,-85.3),\n');
fprintf(mymap,'   icon: pinImageR,\n');
fprintf(mymap,'   map: map\n');
fprintf(mymap,'})');

footer = fopen(footerFile);
while ~feof(footer)
  footerLine = fgetl(footer);
  fprintf(mymap,'%s\n',footerLine);
end
fclose(footer);


fclose(mymap);

File: labs/google_maps_files/make_google_map.py
#!/usr/local/bin/python

headerFile = "header.txt"
footerFile = "footer.txt"

outputFile = "mymap.html"

with open(headerFile, 'r') as header:
    headerLines = header.read()

    header.close()
    
    with open(footerFile, 'r') as footer:
        footerLines = footer.read()

        footer.close()
        
        with open(outputFile, 'w') as mymap:

            # If you want to get super fancy, you can search the headerLines to
            # find the "center" point and change that, so that your map will
            # be centered perfectly, but you don't have to.

            mymap.write(headerLines)

            #-------------------------------------------------------------------
            # Your code goes here !!!
            
            mymap.write("        var marker = new google.maps.Marker({\n")
            mymap.write("           position: new google.maps.LatLng(42.3,-85.3),\n")
            mymap.write("           icon: pinImageR,\n")
            mymap.write("           map: map\n")
            mymap.write("        });\n")

            #-------------------------------------------------------------------
            
            mymap.write(footerLines)

            mymap.close()


            
